jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/enricher/WildflyJARHealthCheckEnricherTest.java

```diff
@@ -81,64 +81,120 @@ private void setupExpectations(Map<String, Map<String, Object>> jkubeConfig) {
-        assertEquals("/foo/live", probe.getHttpGet().getPath());
-        assertEquals(99, probe.getInitialDelaySeconds().intValue());
-        assertEquals(27, probe.getSuccessThreshold().intValue());
-        assertEquals(17, probe.getFailureThreshold().intValue());
-        probe = enricher.getReadinessProbe();
-        assertNotNull(probe);
-        assertEquals("HTTPS", probe.getHttpGet().getScheme());
-        assertNull(probe.getHttpGet().getHost());
-        assertEquals(1234, probe.getHttpGet().getPort().getIntVal().intValue());
-        assertEquals("/foo/ready", probe.getHttpGet().getPath());
-        assertEquals(77, probe.getInitialDelaySeconds().intValue());
-        assertEquals(27, probe.getSuccessThreshold().intValue());
-        assertEquals(17, probe.getFailureThreshold().intValue());
+        final Probe livenessProbe = enricher.getLivenessProbe();
+        assertProbesAdded(livenessProbe, "HTTPS", "/foo/live", 1234, Arrays.asList(99, 3, 1, 10));
+
+        final Probe readinessProbe = enricher.getReadinessProbe();
+        assertProbesAdded(readinessProbe, "HTTPS", "/foo/ready", 1234, Arrays.asList(77, 3, 1, 30));
+
+        final Probe startupProbe = enricher.getStartupProbe();
+        assertThat(startupProbe).isNull();
+    }
+
+    @Test
+    public void testWithCustomConfigurationComingFromConf_withWildflyJarAfter25_0shouldAdd_startupProbe(@Mocked final JavaProject project) {
+        wildFlyJarDependencyWithVersion(project, "26.1.1.Final");
+        Map<String, Object> jarConfig = new HashMap<>();
+        jarConfig.put("cloud", null);
+        final Map<String, Map<String, Object>> config = createFakeConfig(
+                 "{\"readinessPath\":\"/foo/ready\","
+                + "\"livenessPath\":\"/foo/live\","
+                + "\"startupPath\":\"/foo/started\","
+                + "\"port\":\"1234\","
+                + "\"scheme\":\"https\","
+                + "\"livenessInitialDelay\":\"99\","
+                + "\"livenessFailureThreshold\":\"3\","
+                + "\"livenessSuccessThreshold\":\"1\","
+                + "\"livenessPeriodSeconds\":\"10\","
+                + "\"readinessInitialDelay\":\"77\","
+                + "\"readinessFailureThreshold\":\"3\","
+                + "\"readinessSuccessThreshold\":\"1\","
+                + "\"readinessPeriodSeconds\":\"30\","
+                + "\"startupInitialDelay\":\"10\","
+                + "\"startupFailureThreshold\":\"3\","
+                + "\"startupSuccessThreshold\":\"1\","
+                + "\"startupPeriodSeconds\":\"10\""
+                + "}");
+        setupExpectations(project, jarConfig, config);
+
+        final WildflyJARHealthCheckEnricher enricher = new WildflyJARHealthCheckEnricher(context);
+        final Probe livenessProbe = enricher.getLivenessProbe();
+        assertProbesAdded(livenessProbe, "HTTPS", "/foo/live", 1234, Arrays.asList(99, 3, 1, 10));
```

Comment:
This is not very readable, it's hard to say what the `99, 3, 1, 10` values mean.

Especially considering that these values are repeated:
A simple replacement of the Config.READINESS_SUCCESS_THRESHOLD with the Config.STARTUP_SUCCESS_THRESHOLD in the `getReadinessProbe()` wouldn't make this test fail.

Also, some of these values are just the default, value, so a change in the first argument of the Config enum for READINESS_SUCCESS_THRESHOLD wouldn't make the test fail either.